File: Array/56.merge_intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        intervals.sort(key=lambda x: x[0])
        ans = []
        if len(intervals) == 0:
            return ans
        ans.append(intervals[0])
        if len(intervals) == 1:
            return ans
        for i in range(1, len(intervals)):
            inter = ans[-1]
            candidate = intervals[i]
            if candidate[0] > inter[1]:
                ans.append(candidate)
            else:
                ans[-1] = [inter[0], max(inter[1], candidate[1])]
        return ans

File: C++/Sorting/56. Merge Intervals.cpp
#include <algorithm>

class Solution {
public:
    vector<vector<int>> merge(vector<vector<int>>& intervals) {
        sort(intervals.begin(), intervals.end());
        vector<vector<int>> res;
        vector<int> currVec = {-1, -1};
        for (auto& v : intervals) {
            if (v[0] > currVec[1]) {
                if (currVec[0] != -1) res.push_back(currVec);
                currVec = v;
            }
            else {
                currVec[1] = max(currVec[1], v[1]);
            }
        }
        if (currVec[0] != -1) res.push_back(currVec);
        return res;
    }
};